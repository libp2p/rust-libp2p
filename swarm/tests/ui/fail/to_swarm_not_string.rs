use libp2p_ping as ping;

#[derive(libp2p_swarm::NetworkBehaviour)]
#[behaviour(out_event = FooEvent, prelude = "libp2p_swarm::derive_prelude")]
struct Foo {
    ping: ping::Behaviour,
}

struct FooEvent;

impl From<ping::Event> for FooEvent {
    fn from(_: ping::Event) -> Self {
        unimplemented!()
    }
}

fn main() {

}
