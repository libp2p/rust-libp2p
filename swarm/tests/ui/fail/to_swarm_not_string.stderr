error: expected a string literal
 --> tests/ui/fail/to_swarm_not_string.rs:4:25
  |
4 | #[behaviour(out_event = FooEvent, prelude = "libp2p_swarm::derive_prelude")]
  |                         ^^^^^^^^
